hw_diag/diagnostics/key_diagnostics.py

```diff
@@ -24,12 +26,18 @@ def perform_test(self, diagnostics_report):
         # Try to get key, but there may be ECC lock or other failure
         try:
             public_keys = get_public_keys_rust()
+        except ECCMalfunctionException as e:
+            diagnostics_report.record_failure(e, self)
         except UnboundLocalError as e:
-            diagnostics_report.record_failure(str(e), self)
+            diagnostics_report.record_failure(e, self)
         except (FileNotFoundError, NotADirectoryError) as e:
-            diagnostics_report.record_failure(str(e), self)
+            diagnostics_report.record_failure(e, self)
+        except ResourceBusyError as e:
+            diagnostics_report.record_failure(e, self)
         except Exception as e:
-            diagnostics_report.record_failure(str(e), self)
+            diagnostics_report.record_failure(e, self)
+            raise ECCMalfunctionException(e)\
```

Comment:
We never want a diagnostic to throw an exception. This will cause the /initFile.txt endpoint to fail when we want it to be successful but return a payload that indicates some specific component is failing.